server/gitrest/src/routes/git/refs.ts

```diff
@@ -151,53 +29,73 @@ function getRefId(id): string {
+            request.params.owner,
+            request.params.repo,
+        );
+        const resultP = repoManager.getRefs();
         handleResponse(resultP, response);
     });
 
-    router.get("/repos/:owner/:repo/git/refs/*", (request, response, next) => {
-        const resultP = getRef(
-            repoManager,
+    router.get("/repos/:owner/:repo/git/refs/*", async (request, response, next) => {
+        const repoManager = await repoManagerFactory.open(
             request.params.owner,
             request.params.repo,
+        );
+        const resultP = repoManager.getRef(
             getRefId(request.params[0]),
-            utils.getReadParams(request.query?.config),
-            externalStorageManager);
+            getExternalWriterParams(request.query?.config as string),
+        );
         handleResponse(resultP, response);
     });
 
-    router.post("/repos/:owner/:repo/git/refs", (request, response, next) => {
-        const resultP = createRef(
-            repoManager,
+    router.post("/repos/:owner/:repo/git/refs", async (request, response, next) => {
+        const repoManager = await repoManagerFactory.open(
             request.params.owner,
             request.params.repo,
-            request.body as ICreateRefParamsExternal,
-            externalStorageManager);
+        );
+        const createRefParams = request.body as ICreateRefParamsExternal;
+        const resultP = repoManager.createRef(
+            createRefParams,
+            createRefParams.config,
+        );
         handleResponse(resultP, response, 201);
     });
 
-    router.patch("/repos/:owner/:repo/git/refs/*", (request, response, next) => {
-        const resultP = patchRef(
-            repoManager,
+    router.patch("/repos/:owner/:repo/git/refs/*", async (request, response, next) => {
+        const repoManager = await repoManagerFactory.open(
             request.params.owner,
             request.params.repo,
+        );
+        const patchRefParams = request.body as IPatchRefParamsExternal;
+        const resultP = repoManager.patchRef(
             getRefId(request.params[0]),
-            request.body as IPatchRefParamsExternal,
-            externalStorageManager);
+            patchRefParams,
+            patchRefParams.config,
+        );
         handleResponse(resultP, response);
     });
 
-    router.delete("/repos/:owner/:repo/git/refs/*", (request, response, next) => {
-        const deleteP = deleteRef(repoManager, request.params.owner, request.params.repo, getRefId(request.params[0]));
-        deleteP.then(() => response.status(204).end(), (error) => response.status(400).json(error));
+    router.delete("/repos/:owner/:repo/git/refs/*", async (request, response, next) => {
+        const repoManager = await repoManagerFactory.open(
+            request.params.owner,
+            request.params.repo,
+        );
+        const resultP = repoManager.deleteRef(getRefId(request.params[0]));
+        resultP.then(
```

Comment:
Any reason why we don't use `handleResponse` here and elsewhere in the routes where we are currently using `resultP.then(()=>{ ... }, (error)=>{ ... })`

Reply:
Just now made this exact change everywhere to help me debug the test failures :)